account/verification_code.go

```diff
@@ -0,0 +1,207 @@
+package account
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-auth/application/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormsupport"
+	"github.com/fabric8-services/fabric8-auth/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+type VerificationCode struct {
+	gormsupport.Lifecycle
+	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	User     User
+	Verified bool // The verification status of the updated email.
```

Comment:
We don't really need verified field, do we?

Reply:
In the event where we don't hard-delete ( my initial thought as per https://github.com/fabric8-services/fabric8-auth/pull/227#discussion_r157127249 ) , this field would be needed to ensure one doesn't use the same confirmation url twice.

Reply:
He couldn't because the code would be soft deleted, could he?

Reply:
Right - now that we decided to soft-delete. 
I will remove the `verified` field.